File: sysprogramming/lab1/lab1.go
package main

import (
	"fmt"
	"io"
	"log"
	"os"
	"strings"
)

func main() {
	var inputPath = os.Args[1]
	var outPath = os.Args[2]
	dentry, err := os.ReadDir(inputPath)
	if err != nil {
		log.Println(err)
		os.Exit(1)
	}
	outFile, err := os.OpenFile(outPath, os.O_RDWR|os.O_CREATE, 0644)
	if err != nil {
		panic(err)
	}
	defer outFile.Close()
	for _, entry := range dentry {
		if entry.Name() == ".." || entry.Name() == "." || entry.IsDir() {
			continue
		}

		file, err := os.Open(inputPath + entry.Name())
		if err != nil {
			log.Println(err)
			os.Exit(2)
		}
		defer file.Close()
		data, err := io.ReadAll(file)
		if err != nil {
			log.Println(err)
			os.Exit(3)
		}
		count := strings.Count(string(data), " ") +
			strings.Count(string(data), "\t")
		io.WriteString(outFile, fmt.Sprintf("%s -> %d\n", inputPath+entry.Name(), count))
	}
}

File: sysprogramming/lab1/lab1.zig
const std = @import("std");
const fs = std.fs;
const linux = std.os.linux;

pub fn main() !void {
    var dir = try fs.cwd().openIterableDir("input", .{});
    defer dir.close();
    var it = dir.iterate();
    var output = try fs.cwd().openFile("output/output", .{ .mode = .write_only });
    defer output.close();
    while (try it.next()) |entry| {
        var file = try dir.dir.openFile(entry.name, .{ .mode = .read_write });
        defer file.close();
        var data = try file.readToEndAlloc(std.heap.page_allocator, 65535);
        var spaces = std.mem.count(u8, data, " ");
        var tabs = std.mem.count(u8, data, "\t");
        var blancs = spaces + tabs;
        try std.fmt.format(output.writer(), "input/{s} -> {d}\n", .{ entry.name, blancs });
    }
}
